search/FloydAlgorithmForCycleDetection.cpp

```diff
@@ -0,0 +1,42 @@
+
+/*Given an array of integers nums containing n + 1 integers where each integer is in the range [1, n] inclusive.
+
+There is only one duplicate number in nums, return this duplicate number in O(1) space and time complexity less than O(n^2) without modifying the original array.
+
+Example-:
+    Input: 5
+           3 1 3 4 2
+    
+    Output: 3
+*/
+#include<bits/stdc++.h>
+using namespace std;
+    int findDuplicate(vector<int>& nums) {
+        int slow=nums[0];

```

Comment:
you have declared c style arrays, use std::array, for example see [https://github.com/TheAlgorithms/C-Plus-Plus/pull/1085](url)

Reply:
bits/stdc++ library is not allowed, as it is Linux specific and it also slows down compilation process. so include only required libraries.